Source/ARTAuth.m

```diff
@@ -391,6 +391,26 @@ - (void)requestToken:(ARTTokenParams *)tokenParams withOptions:(ARTAuthOptions *
     } else {
         void (^tokenDetailsFactory)(ARTTokenParams *, void(^)(ARTTokenDetails *_Nullable, NSError *_Nullable));
         if (replacedOptions.authCallback) {
+            // We're giving this callback to the user, and we don't control their code. If they were to hold on to
+            // this callback, they could introduce a leak, since the callback may hold strong references
+            // to Rest/Realtime.
+            //
+            // So we're going to apply the Fundamental Theorem of Software Engineering here and wrap the
+            // callback in another reference. Then, we pass a callback that calls this reference,
+            // and not the original callback. Then, we return a task that, if cancelled, sets that
+            // intermediate reference to nil, so that we lose our reference to the original callback
+            // and release any references it may hold to Rest/Realtime. Even if the user holds on to
+            // the callback we give to them, they would be then holding to a now-nil reference to the
+            // original callback.
+            __weak void (^safeCallback)(id<ARTTokenDetailsCompatible> tokenDetailsCompat, NSError * error);
```

Comment:
Why is this weak? Which reference cycle is this breaking?

Reply:
Just being defensive as I don't think the current scope has any business in 'owning' that callback. On other words, erring on the side of caution, in which case my question in return would be, why _should_ it be strong? It feels like a delegation pattern.